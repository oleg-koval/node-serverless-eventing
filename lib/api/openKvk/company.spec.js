jest.mock('./config')
jest.mock('request-promised')

const request = require('request-promised')
const getById = require('./company')
const { getApiParams } = require('./config')

describe('Checks if company is active', () => {
    it('should call openKvk and return company', async () => {
        getApiParams.mockReturnValue({
            headers: {
                'ovio-api-key': 'secret',
            },
            json: true,
            uri: 'https://example.com/42',
        })
        const fixtureResponse = {
            body: {
                actief: true,
                handelsnaam: 'test company',
            },
        }
        request.get.mockReturnValue(fixtureResponse)

        const result = await getById('42')
        expect(result).toMatchSnapshot()
    })

    it('should return error if body is not valid', async () => {
        getApiParams.mockReturnValue({
            headers: {
                'ovio-api-key': 'secret',
            },
            json: true,
            uri: 'https://example.com/42',
        })
        const fixtureResponse = {
            body: undefined,
        }
        request.get.mockReturnValue(fixtureResponse)

        try {
            await getById('42')
        } catch (error) {
            expect(error).toMatchSnapshot()
        }
    })
})
