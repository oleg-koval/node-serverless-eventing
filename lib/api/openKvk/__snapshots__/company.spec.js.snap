// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks if company is active should return error if body is not valid 1`] = `[Error: no response from openKvk]`;

exports[`Checks if company is active should return error if missing active information 1`] = `[Error: missing activity information]`;
